apps/gallery/test/marionette/delete_image_test.js

```diff
@@ -19,37 +19,117 @@ marionette('using the gallery menu', function() {
+      });
+    });
   });
 
-  test('should let you delete images', function() {
-    // This test deletes the test image.
-    app.thumbnailsSelectButton.click();
-    app.thumbnail.click();
-    app.thumbnailsDeleteButton.click();
-    app.confirmButton.click();
-
-    // Wait for the overlay with 'No photos or videos' message.
-    var overlayView = app.overlayView;
-    client.helper.waitForElement(overlayView);
-    assert.ok(overlayView.displayed());
+  suite('Gallery app with multiple images', function() {
+    setup(function() {
+      // Remove all files in temp device storage.
+      client.fileManager.removeAllFiles();
+      // Add file into the pictures directory
+      client.fileManager.add([
+        {type: 'pictures',filePath: 'apps/gallery/test/images/01.jpg'},
+        {type: 'pictures',filePath: 'apps/gallery/test/images/02.png'},
+        {type: 'pictures',filePath: 'apps/gallery/test/images/03.gif'}
+      ]);
+    });
+
+    suite('Select and delete multiple images', function() {
+      setup(function() {
+        app.launch();
+      });
+
+      test('should let you select images', function() {
+        // Use the 'selection' button and select the test images.
+        // Selected images should have an border/highlight.
+        app.thumbnailsSelectButton.click();
+        app.tapThumbnail(0);
+        app.tapThumbnail(1);
+        app.tapThumbnail(2);
+
+        assert.ok(app.thumbnails[0].cssProperty('outline') != null);
+        assert.ok(app.thumbnails[1].cssProperty('outline') != null);
+        assert.ok(app.thumbnails[2].cssProperty('outline') != null);
+
+        assert.ok(app.thumbnailsNumberSelected.text() == '3 selected');
+
+      });
+
+      test('should let you cancel delete images', function() {
+        // Use the 'selection' button and select the test images.
+        // Selected images should have an border/highlight.
+        app.thumbnailsSelectButton.click();
+        app.tapThumbnail(0);
+        app.tapThumbnail(1);
+        app.tapThumbnail(2);
+        // Check number of selected images
+        assert.ok(app.thumbnailsNumberSelected.text() == '3 selected');
+        app.thumbnailsDeleteButton.click();
+        // Cancel delete of images
+        app.confirmCancelButton.click();
+
+        // Check files still exist
+        assert.ok(app.thumbnails.length == '3');
+        assert.ok(app.thumbnailsNumberSelected.text() == '3 selected');
+      });
+
+      test('should let you delete images', function() {
+        // This test deletes the test image.
+        app.thumbnailsSelectButton.click();
+        app.tapThumbnail(0);
+        app.tapThumbnail(1);
+        app.tapThumbnail(2);
+
+        app.thumbnailsDeleteButton.click();
+        app.confirmButton.click();
+
+        // Wait for the overlay with 'No photos or videos' message.
+        var overlayView = app.overlayView;
+        client.helper.waitForElement(overlayView);
+        assert.ok(overlayView.displayed());
```

Comment:
I'd suggest that you also assert that app.thumbnails.length === 0.  The overlay thing seems like a special case